File: 준영/LV2/캐시.py
def solution(cacheSize, cities):
    answer = 0

    city_cache = []

    for city in cities:
        if cacheSize == 0:
            answer = 5 * len(cities)
            break

        if city.lower() not in city_cache:
            if len(city_cache) < cacheSize:
                city_cache.append(city.lower())
            else:
                city_cache.pop(0)
                city_cache.append(city.lower())    
            answer += 5
        else:
            n = city_cache.index(city.lower())
            city_cache.pop(n)
            city_cache.append(city.lower())    
            
            answer += 1
    return answer


cacheSize = 3
cities = ["Jeju", "Pangyo", "Seoul", "NewYork", "LA", "Jeju", "Pangyo", "Seoul", "NewYork", "LA"]
solution(cacheSize, cities)

File: 준영/프로그래머스/LV2/캐시.js
function solution(cacheSize, cities) {
  var answer = 0
  const city_cache = []

  for (const city of cities) {
    if (cacheSize === 0) {
      answer = 5 * cities.length
      break
    }

    var new_city = city.toLowerCase()

    if (!city_cache.includes(new_city)) {
      if (city_cache.length < cacheSize) {
        city_cache.push(new_city)
      } else {
        city_cache.shift()
        city_cache.push(new_city)
      }
      answer += 5
    } else {
      const n = city_cache.indexOf(new_city)
      city_cache.splice(n, 1)
      city_cache.push(new_city)
      answer += 1
    }
    
  }
  return answer
}